theano/tensor/basic.py

```diff
@@ -4372,7 +4372,7 @@ def infer_shape(self, node, ishapes):
                     if ele == -1:
                         requ[i] = missing
             elif crit == 1:  # we reshape to -1
-                requ = [mul(*ishapes[0])]
+                requ = [mul(*ishapes[0])] if ishapes[0] else [1]
```

Comment:
If the output is a scalar, then yes, it should return [[]].

If this PR need some tests to be commented and that gh-3420 would fix that, we should wait for gh-3420 before merging this PR.